app/models/group.rb

```diff
@@ -31,8 +31,41 @@ def roles
     @roles ||= Group.find(uuid).roles
   end
 
-  def has_role?(role_name)
-    roles.any? { |role| role.name == role_name }
+  def can_approve?
```

Comment:
cache the result
```
def can_approve?
  return @can_approve unless @can_approve.nil?
  @can_approve = ...
end
```

Reply:
Should just be able to use `@can_approval ||=`